yarn-project/foundation/src/json-rpc/server/json_rpc_server.ts

```diff
@@ -98,7 +117,15 @@ export class JsonRpcServer {
             // Propagate the error message to the client. Plenty of the errors are expected to occur (e.g. adding
             // a duplicate recipient) so this is necessary.
             ctx.status = 400;
-            ctx.body = { error: err.message };
+            ctx.body = {
+              jsonrpc,
+              id,
+              error: {
+                // sent back an error code outside the reserved range
+                code: 1,
```

Comment:
Do you think we would want to use here [implementation-defined server-errors](https://www.jsonrpc.org/specification#error_object)?

I think it would be a nice thing to have.

If you agree please create the issue and add a comment next to the code with TODO #issue_number and we'll tackle that in a separate PR.

Reply:
I think that's a good idea, but won't be easy to implement because errors can be thrown by different components in the system :thinking:. I think we should still do it, but in the meantime maybe a different error code would be more appropriate here? I see Ethereum uses `-32000` which is the first of the 'server reserved range' as a default error code.

Reply:
> but won't be easy to implement because errors can be thrown by different components in the system

I think we could just extend an error like this:
`class AztecError extends Error {

  constructor(public readonly code: number, message: string) {
    super(message);
  }
}`
and then iterate through all the occurrences of `throw new Error` in our codebase and use it there.

Reply:
Yep! I think though that it would make sense to have an enum for error codes so that RPC clients don't end up having to handle the same error code for two different types of errors, e.g.:

```ts
enum ErrorCode {
  InvalidAztecAddress = 10000
}

// in some API
throw new AztecError(ErrorCode.InvalidAztecAddress, "invalid address")
```